zerver/models.py

```diff
@@ -2884,7 +2881,7 @@ def get_stream_by_id_in_realm(stream_id: int, realm: Realm) -> Stream:
     return Stream.objects.select_related().get(id=stream_id, realm=realm)
 
 
-def bulk_get_streams(realm: Realm, stream_names: STREAM_NAMES) -> Dict[str, Any]:
+def bulk_get_streams(realm: Realm, stream_names: Set[str]) -> Dict[str, Any]:
```

Comment:
I think we're supposed to use some other ABC collection rather than Set, but I agree the old code was excessive.

Reply:
I really prefer to require concrete types when possible.  The caller can always set-ify things as needed.

If the concern is enforcing immutability, I really, really hate our current paradigm.  Just call an immutable set "ImmutableSet".  Even if we have to typevar it to "Collection" or some other thing.  "Collection" is so completely vague.  What are its semantics?  Is it even immutable? I have no idea.